libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -819,6 +844,33 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 			solAssert(false, "Invalid member access in contract");
 		break;
 	}
+	case Type::Category::TypeType:
```

Comment:
This part should already be fully implemented in https://github.com/ethereum/solidity/pull/8685

Reply:
Yeah, I saw it. In my case this is just a no-op for the cases I'm adding support for and a couple of `solUnimplementedAssert`s to explicitly mark cases that I know are not implemented yet. I don't strictly need this code (an empty `case Type::Category::TypeType: break;` would get the job done too) but I thought that having more sanity checks wouldn't hurt. Can I leave it as is?

Alternatively I can try to make it a bit closer to what you have in #8685 to make combining our code easier.

Reply:
#8685 already went in so in the end I dropped my code for this part. Most of the checks are already covered, just done a bit differently.